### PR TITLE
add minijinja_contrib dependency

### PR DESCRIPTION
**Pull request by cai-n**
add minijinja_contrib dependency to support datetime filters.

### CODE REVIEW

File: crates/kumod/src/logging.rs [hunk 1]
```diff
@@ -197,6 +201,10 @@ impl Logger {
 
         for (kind, per_rec) in &params.per_record {
             if let Some(template_source) = &per_rec.template {
+                if let Some(datetime_format) = &per_rec.template_datetime_format {
+                    template_engine.add_global("DATETIME_FORMAT", datetime_format.clone());
+                }
```
Comment (wez) [inline crates/kumod/src/logging.rs]: I'm not so sure about this part; here's what's on my mind:

* If we add this, it suggests that we also should add the other global variables mentioned in the docs; `DATE_FORMAT`, `TIME_FORMAT` and `TIMEZONE`
* Setting up those optional globals would need to happen in each place where we configure minijinja; there are two locations in this file, but we also use it for the HTTP injection API, and any other place where we add templating in the future
* It feels like a lot of repetition for an optional feature.

I wonder if we can simply skip this and leave it to jinja syntax.  My understanding is that you should be able to set global variables at the top of your template block, and then use the new filter like this:

```jinja
{% set DATETIME_FORMAT = "iso" %}
created={{ created | datetimeformat }}
when={{ timestamp | datetimeformat }}
```

alternatively, although slightly more verbose, this doesn't seem too bad:

```jinja
created={{ created | datetimeformat(format="iso") }}
when={{ timestamp | datetimeformat(format="iso") }}
```

what do you think about this? Can you share more information about what motivated you to do it this way?

**Comment by wez** [comment]
Please also enable these new functions for the injection API over here: https://github.com/KumoCorp/kumomta/blob/ffc15f2f74e358c68701e7a947a8f95ca8033e70/crates/kumod/src/http_server/inject_v1.rs#L237
Thanks!

File: crates/kumod/src/logging.rs (hunk 1 above)
Comment (cai-n) [reply crates/kumod/src/logging.rs]: Ah yeah, haven't though about setting the global in jinja directly, that totally works.
`template = '{%- set DATETIME_FORMAT = "[year]-[month]-[day] [hour]:[minute]:[second][offset_hour sign:mandatory][offset_minute]" -%}{{ type[:1] | lower }},{{ timestamp | datetimeformat }}'`
This gives me exactly what I'll looking for

**Comment by wez** [comment]
Thanks!

**Pull request merged**